### PR TITLE
Rename backend Pattern cases from PBlank to MPBlank style

### PR DESCRIPTION
**Pull request by StachuDotNet**
Changelog:

```
Internal
- rename backend Match Pattern cases from "PBlank"-style to "MPBlank" style in preparation for Let Patterns
```

This is part of #4489.

Match Patterns (the things we use to `match` against) were previously branded as just Patterns. In transitioning from Patterns to Match Patterns, we had [previously](#4487) updated the `client` halfway to support Match Pattern cases serializing as "MPBlank" rather than "PBlank." The types were changed, and the `decoder` was updated to handle both styles of naming.

This PR updates the backend's ClientTypes in the same way - everything is branded "Match Pattern" internally, with compatibility for both in the 'decoder.'

After this PR, the _encoders_ of both the backend and the client still use the old PVariable-style naming.

The final step here will be to remove the old-style naming convention from both client and fsharp-backend. The types, encoder/decoder, etc. can all be done in one PR in a week or so once users' clients are up to date. Several Vanilla-serialized .json files used for backend testing will need updated as part of this.

### CODE REVIEW

**Comment by StachuDotNet** [comment]
I forgot - there are still a few serializable things not entirely in `ClientTypes` space.

Two that popped up after (only) updating the naming style of PT Patterns:
- loading a Canvas from a .json file on disk (used for integration tests only?)
- cloning a Canvas (used by F# unit tests only?)

**Comment by StachuDotNet** [comment]
> loading a Canvas from a .json file on disk (used for integration tests only?)

This was indeed only used by integration tests. I updated those .json files and think it's fine to just leave them as (internal) PT types for now.

> cloning a Canvas (used by F# unit tests only?)

It turns out that we don't actually Vanilla-serialize/deserialize any "internal" types during canvas-cloning. Rather, the referenced type was only used in testing of canvas-cloning. I've adjusted the test to no longer reference the 'internal' type but to instead roundtrip the CT equivalent.

**Comment by StachuDotNet** [comment]
> It might save some time if you add both MP and non-MP names in clienttypes in this PR. That will allow transitioning the types on the client in a single later deploy.

That's the plan - just taking care of internal namings first

**Comment by StachuDotNet** [comment]
Both client-side and server-side decoders for the ClientTypes are prepared for `MPVariable`-style namings of these cases (in addition to the current/old-style namings).

I would go ahead and update the server-side encoder to prefer the new-style namings, but it causes the Vanilla roundtripping tests to fail.

The final step here is to remove the old-style naming convention from both `client` and `fsharp-backend`. The types, encoder/decoder, etc. can all be done in one fell swoop. Several Vanilla-serialized .json files used for backend testing will need updated as part of this.

**Comment by pbiggar** [comment]
Just replaying this back so I can review:
- client can already read MP and P constructors
- client already uses MP names
- changes:
  - the backend now uses MP everywhere
  - the backend client types can now read both P and MP
- non-changes:
  - encoders both use P for now

Can you confirm my understanding?

**Comment by StachuDotNet** [comment]
Yes, that's all exactly correct.

File: fsharp-backend/src/LibBinarySerialization/SerializedTypes.fs [hunk 1]
```diff
@@ -91,17 +91,17 @@ module FQFnName =
...
-  | PConstructor of id * string * List<Pattern>
-  | PInteger of id * int64
-  | PBool of id * bool
-  | PCharacter of id * string
-  | PString of id * string
-  | PFloat of id * Sign * string * string
-  | PNull of id
-  | PBlank of id
-  | PTuple of id * Pattern * Pattern * List<Pattern>
+type MatchPattern =
```
Comment (pbiggar) [inline fsharp-backend/src/LibBinarySerialization/SerializedTypes.fs:94]: Is renaming the type allowed?

I guess we must have tests, so it must be?

**Review by pbiggar — APPROVED** [review]
Nice

File: fsharp-backend/src/LibBinarySerialization/SerializedTypes.fs (hunk 1 above)
Comment (StachuDotNet) [reply fsharp-backend/src/LibBinarySerialization/SerializedTypes.fs:94]: Yes, it seems to be. The order is what matters afaict

**Pull request merged**